addon/helpers/optional-type.js

```diff
@@ -0,0 +1,8 @@
+import { helper } from "@ember/component/helper";
+import { createUnionOfValidator } from '../-private/validators';
+
+export function optionalType([validator]) {
```

Comment:
I feel having the helper named "optional" would provide better readability when combining this helper with other helpers.

For example: `{{arg-type this.value (optional (array-of  "string"))}}`

Reply:
I was going to go with that but felt we might run into naming collisions. If you're cool with it I'll change it.